### PR TITLE
Auto-upload crashes to backtrace.io

### PR DESCRIPTION
**Pull request by janisozaur**
This is a proof-of-concept of https://backtrace.io integration.

It works using breakpad's native uploader and is capable of automatically creating issues such as https://github.com/OpenRCT2/OpenRCT2/issues/8072 (using the crash reported in https://github.com/OpenRCT2/OpenRCT2/issues/8047).

I have manually uploaded the symbols from a build, it would require some automation, a case of uploading `.pdb` to one more place.

As of now this is done using 30-day trial on backtrace, things to do to make this all work:
* [x] apply for open-source exemption (1000crashes/month, 90 days of retention) see https://documentation.backtrace.io/open_source/
* [ ] provide accounts for other team members
* [x] upload the symbols to backtrace as part of build

I expect some discussion to follow…

### CODE REVIEW

**Comment by janisozaur** [comment]
Our account got upgraded to the open source offering.

File: scripts/ps/appveyor_deploy.ps1 [hunk 1]
```diff
@@ -14,6 +14,11 @@ if ($nottesting -and $notvs2015)
         if (${env:APPVEYOR_REPO_TAG} -eq "true" -or ${env:APPVEYOR_REPO_BRANCH} -match "^develop$|^push/")
         {
             msbuild openrct2.proj /t:UploadArtifacts /logger:"C:\Program Files\AppVeyor\BuildAgent\Appveyor.MSBuildLogger.dll"
+            # PowerShell is being stupid here and hijacks calls to curl, need to call it via curl.exe explicitly
```
Comment (IntelOrca) [inline scripts/ps/appveyor_deploy.ps1]: Please change the comment. PowerShell is not being stupid. There is by default an alias of curl to `Invoke-WebRequest`.

File: src/openrct2/platform/Crash.cpp [hunk 2]
```diff
@@ -39,6 +42,31 @@ const wchar_t* _wszCommitSha1Short = WSZ("");
 // OPENRCT2_ARCHITECTURE is required to be defined in version.h
 const wchar_t* _wszArchitecture = WSZ(OPENRCT2_ARCHITECTURE);
 
+static bool UploadMinidump(const wchar_t* dumpPath)
+{
+    std::wstring url(
+        L"https://submit.backtrace.io/openrct2/e9e6d681fafdeac9f6131b4b59a155d54bebad567a8c0380d70643f4414819f5/minidump");
```
Comment (IntelOrca) [inline src/openrct2/platform/Crash.cpp]: This could be extracted to a constant at the top of the file. What does the long hex string represent?

**Comment by IntelOrca** [comment]
Do the dump files get compressed before being sent to the server? One mini dump I got was ~8 MiB, but compressed it was ~500 KiB. I am just wondering whether we need to show a status box to say it is uploading if it is likely to take awhile.

File: src/openrct2/platform/Crash.cpp (hunk 2 above)
Comment (janisozaur) [reply src/openrct2/platform/Crash.cpp]: Access token

File: scripts/ps/appveyor_deploy.ps1 (hunk 1 above)
Comment (janisozaur) [reply scripts/ps/appveyor_deploy.ps1]: Being stupid by design is still stupid. Hijacking calls to a widely available utility that doesn't even try to mimic original's behaviour is stupid.

**Comment by janisozaur** [comment]
It is not compressed. I can look into what we can do with the tools we already have, but I'd like to limit amount of work being done in an already crashed state.

I can add a note stating the upload my take a while and won't show progress, maybe add a message box once it's done.

**Comment by IntelOrca** [comment]
I wonder if it automatically gets compressed using gzip over HTTP, providing the server returns an accept header for gzip.

**Comment by janisozaur** [comment]
I've updated this with ability to gzip minidumps, but would need someone to test it out.

**Comment by janisozaur** [comment]
I've disabled gzip for now, Backtrace's support said it's likely on their roadmap and will come back to me with confirmation/ETA.

**Comment by janisozaur** [comment]
See https://github.com/OpenRCT2/OpenRCT2/issues/8157 for a crash generated using (manually inserted bug) https://github.com/OpenRCT2/OpenRCT2/pull/8073/commits/76b6bd82a0d27d632dd0c6570947ec610accc664

**Pull request merged**